build.gradle

```diff
@@ -1014,43 +1017,82 @@ task testIntegration(type: JavaExec) {
-}
-
-task javadocJar(type: Jar) {
-    from javadoc.outputs
-    classifier = 'javadoc'
+java {
+    withSourcesJar()
+    withJavadocJar()
 }
 
 ext.mavenStyleVersion = version.replace('_', '-')
 
 publishing {
     publications {
-        bintray(MavenPublication) {
+        mavenJava(MavenPublication) {
+            from(components.java)
             groupId 'org.omegat'
             artifactId 'omegat'
             version mavenStyleVersion
-            from components.java
-            artifact sourceJar
-            artifact javadocJar
+            pom {
+                name = "omegat"
+                description = "OmegaT - the free translation memory tool"
+                url = "https://sourceforge.net/p/omegat"
+                licenses {
+                    license {
+                        name = "The GNU General Public License, Version 3.0"
+                        url = "https://www.gnu.org/licenses/licenses/gpl-3.0.html"
+                        distribution = "repo"
+                    }
+                }
+                developers {
+                    developer {
+                        name = "OmegaT team"
+                        id = "omegat"
+                        email = "developer@omegat.org"
+                    }
+                }
+                scm {
+                    connection = "scm:git:git://git.code.sf.net/p/omegat/code"
+                    developerConnection = "scm:git:git://git.code.sf.net/p/omegat/code.git"
+                    url = "https://sourceforge.net/projects/omegat/"
+                }
+            }
+        }
+    }
+    repositories {
+        maven {
+            name = "GitHubPackages"
+            url = "https://maven.pkg.github.com/omegat-org/omegat"
+            credentials {
+                username = System.getenv("GITHUB_ACTOR")
+                password = System.getenv("GITHUB_TOKEN")
+            }
         }
     }
 }
 
-bintray {
-    user = findProperty('bintrayUser')
-    key = findProperty('bintrayKey')
-    pkg {
-        repo = 'maven'
-        name = 'OmegaT'
-        userOrg = 'omegat-org'
-        version {
-            name = mavenStyleVersion
+nexusPublishing {
+    repositories {
+        sonatype {
+            nexusUrl.set(uri("https://s01.oss.sonatype.org/service/local/"))
+            snapshotRepositoryUrl.set(uri("https://s01.oss.sonatype.org/content/repositories/snapshots/"))
+            username = project.hasProperty('sonatypeUsername') ? project.property('sonatypeUsername') : System.getenv('SONATYPE_USER')
+            password = project.hasProperty('sonatypePassword') ? project.property('sonatypePassword') : System.getenv('SONATYPE_PASS')
         }
     }
-    publications = ['bintray']
+}
+
+signing {
+    def signingKey = findProperty("signingKey")
+    def signingPassword = findProperty("signingPassword")
+    if (signingKey) {
+        useInMemoryPgpKeys(signingKey, signingPassword)
+    } else {
+        useGpgCmd()
```

Comment:
It assumes developer uses GnuPG for singing when manual operation, and set `signing.gnupg.keyName`
property.

Developer can set `signingKey` property which is a ascii-armor export of private key itself.
You can obtain  it by `gpg --export-secret-keys -a <keyid>`

When CI process a key,
```
    env:
        ORG_GRADLE_PROJECT_signingKey: ${{ secrets.GPG_SECRET_KEY }}
        ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.GPG_KEY_PASSPHARASE }}
```
You can set it through enviromnent variables.